lib/kcerts/keys.go

```diff
@@ -0,0 +1,158 @@
+package kcerts
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"golang.org/x/crypto/ssh"
+	"strings"
+)
+
+type PrivateKey struct {
+	Key PrivateKeyProvider
+}
+
+func (e PrivateKey) Signer() (ssh.Signer, error) {
+	res, err := NewSSHSigner(e.Key.Signer(), e.Key.SigningAlgo())
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+type PrivateKeyProvider interface {
+	Signer() crypto.Signer
+	SigningAlgo() string
+	Raw() interface{}
+	SSHPemEncode() ([]byte, error)
```

Comment:
why use PEM encoding at all at this point? ssh has its own marshal / unmarshal, that seems to work just fine, and does not land us in uncharted territory (eg, ED25519 not having a PEM supported encoding, ssh-keygen -m PEM -e ed25519 key fails).

Reply:
Assuming it works (and I believe it was working for me, in previous PR - it just required a new server release to not have the PEM encoding in the way).

Reply:
ssh only has marshalling for public keys, not private. This is an interface for private